Core/Core/SwiftUIViews/ChangeObserver.swift

```diff
@@ -0,0 +1,61 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2021-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public extension View {
+
+    /**
+     iOS 13 implementation of the `onChange` method which is iOS 14 only. On iOS 14 this method directly uses `onChange`.
+     */
+    func onDataChange<Value: Equatable>(of value: Value, perform action: @escaping (_ newValue: Value) -> Void) -> some View {
+        SwiftUI.Group {
+            if #available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *) {
+                self.onChange(of: value, perform: action)
+            } else {
+                ChangeObserver(value: value, action: action) {
+                    self
+                }
+            }
+        }
+    }
+}
+
```

Comment:
since this is only for ios 13, we should add an @available(iOS, obsoleted: 14) as a reminder

Reply:
Good idea, added.